From 01fdb170339e60a59a2d247edca95cce646fa4d7 Mon Sep 17 00:00:00 2001
From: Oliver Giles <ohw.giles@gmail.com>
Date: Tue, 29 May 2018 21:20:40 +0300
Subject: [PATCH] Check size of space allocated for Date object

Check there is enough memory allocated for the Date object
before rewriting it
---
 lib/ytnef.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/ytnef.c b/lib/ytnef.c
index 548fcfa..af96c31 100644
--- a/lib/ytnef.c
+++ b/lib/ytnef.c
@@ -655,6 +655,9 @@ int TNEFDateHandler STD_ARGLIST {
   WORD * tmp_src, *tmp_dst;
   int i;
 
+  if (size < sizeof(dtr))
+    return -1;
+
   p = &(TNEF->starting_attach);
   switch (TNEFList[id].id) {
     case attDateSent: Date = &(TNEF->dateSent); break;
